aws/resource_aws_launch_template_test.go

```diff
@@ -1077,6 +1125,80 @@ resource "aws_launch_template" "test" {
 }
 `
 
+const testAccAWSLaunchTemplateConfig_associatePublicIpAddressTrue = `
```

Comment:
Nit: This can be simplified into a single test configuration function by passing in the parameter, e.g.

```go
func testAccAWSLaunchTemplateConfig_associatePublicIpAddress(associatePublicIpAddress string) string {
  return fmt.Sprintf(`
# ...
resource "aws_launch_template" "test" {
  name = "network-interface-launch-template"

  network_interfaces {
    associate_public_ip_address = %[1]s
    network_interface_id        = "${aws_network_interface.test.id}"
    ipv4_address_count          = 2
  }
}
`, associatePublicIpAddress)
```

Where that is populated via:

```go
Config: testAccAWSLaunchTemplateConfig_associatePublicIpAddress("true"),
Config: testAccAWSLaunchTemplateConfig_associatePublicIpAddress("false"),
Config: testAccAWSLaunchTemplateConfig_associatePublicIpAddress("null"),
```